src/jit/morph.cpp

```diff
@@ -14057,6 +14057,12 @@ GenTree* Compiler::fgMorphModToSubMulDiv(GenTreeOp* tree)
         denominator = fgMakeMultiUse(&tree->gtOp2);
     }
 
+    if (lvaLocalVarRefCounted)
```

Comment:
Yes, I think the ref counts work out in the non-leaf case, though as you say it's a waste. However, in the leaf case, I *think* it is the case that the cloning will cause the ref counts to be incremented, and the original node will be in its original place and already ref-counted. So there will be an extra increment when we call `lvaRecursiveIncRefCounts`. I don't think we have any checks that ref counts are accurate - only that that don't go below zero, so we rarely catch accidental increments. Nevertheless, it makes the heuristics less effective if they are inaccurate.

---
In reply to: [94657932](https://github.com/dotnet/coreclr/pull/8804#discussion_r94657932) [](ancestors = 94657932)